src/main/java/no/nav/familie/integrasjoner/arbeidsfordeling/ArbeidsfordelingClient.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.familie.integrasjoner.arbeidsfordeling
+
+import no.nav.familie.integrasjoner.felles.OppslagException
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.binding.ArbeidsfordelingV1
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.ArbeidsfordelingKriterier
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.Diskresjonskoder
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.Geografi
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.informasjon.Tema
+import no.nav.tjeneste.virksomhet.arbeidsfordeling.v1.meldinger.FinnBehandlendeEnhetListeRequest
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Service
+
+
+@Service
+class ArbeidsfordelingClient(
+        private val arbeidsfordelingV1: ArbeidsfordelingV1
+) {
+
+    fun ping() {
+        arbeidsfordelingV1.ping()
+    }
+
+    fun finnBehandlendeEnhet(gjeldendeTema: String, gjeldendeGeografiskOmråde: String?, gjeldendeDiskresjonskode: String?): List<Enhet> {
+        val request = FinnBehandlendeEnhetListeRequest().apply {
+            arbeidsfordelingKriterier = ArbeidsfordelingKriterier().apply {
+                tema = Tema().apply { value = gjeldendeTema }
+
+                if (gjeldendeGeografiskOmråde != null) {
+                    geografiskTilknytning = Geografi().apply {
+                        value = gjeldendeGeografiskOmråde
+                    }
+                }
+
+                if (gjeldendeDiskresjonskode != null) {
+                    diskresjonskode = Diskresjonskoder().apply {
+                        value = gjeldendeDiskresjonskode
+                    }
+                }
+            }
+        }
+
+        return Result.runCatching { arbeidsfordelingV1.finnBehandlendeEnhetListe(request) }
+                .map { it.behandlendeEnhetListe.map { Enhet(it.enhetId, it.enhetNavn) } }
```

Comment:
Nøstet lamdba med it som parameter i begge lambdaene gjør det lett å blande sammen hvilken it som er hva. Den indre eller ytre lambdaen bør få et annet navn på parameteren sin.